include/ltest-macros.lfe

```diff
@@ -1,62 +1,101 @@
+;; Include EUnit macros
 (include-lib "eunit/include/eunit.hrl")
 
+;;;===================================================================
+;;; OTP 18 hacks
+;;;===================================================================
+
+(defmacro otp-18? () `(=:= "18" (erlang:system_info 'otp_release)))
+
+(defmacro assertion-failed () `(if (otp-18?) 'assert 'assertion_failed))
+
+(defmacro assert-equal-failed ()
+  `(if (otp-18?) 'assertEqual 'assertEqual_failed))
+
+(defmacro assert-not-equal-failed ()
+  `(if (otp-18?) 'assertNotEqual 'assertNotEqual_failed))
+
+(defmacro assert-exception-failed ()
+  `(if (otp-18?) 'assertException 'assertException_failed))
+
+(defmacro assert-match-failed ()
+  `(if (otp-18?) 'assertMatch 'assertMatch_failed))
+
+
+;;;===================================================================
+;;; Helper functions
+;;;===================================================================
+
 (eval-when-compile
   (defun to-unders (atm)
     (re:replace (atom_to_list atm) "-" "_" '(#(return list) global)))
   (defun list-body
-    ((body) (when (is_list body))
-      body)
-    ((body)
-      (list body)))
+    ([body] (when (is_list body))
+     body)
+    ([body]
+     (list body)))
   ;; end of eval-when-compile
   )
 
-(defmacro deftest arg
-  "This macro is for defining standard EUnit tests."
-  (let ((name (car arg))
-        (body (cdr arg)))
-    `(defun ,(list_to_atom (++ (atom_to_list name) "_test")) ()
-       ,@body)))
-
-(defmacro deftestgen arg
-  "This macro is for defining EUnit tests that use test generators."
-  (let ((name (to-unders (car arg)))
-        (body (cdr arg)))
-    `(defun ,(list_to_atom (++ name "_test_")) ()
-       ,@body)))
-
-(defmacro deftestskip arg
-  "This macro is for defining standard EUnit test that will be skipped (not
-  run)."
-  (let ((name (car arg))
-        (body (cdr arg)))
-    `(defun ,(list_to_atom (++ (atom_to_list name) "_skip")) ()
-       ,@body)))
+
+;;;===================================================================
+;;; Test definition macros
+;;;===================================================================
+
+(defmacro deftest
+  "Define a standard EUnit test."
+  ([name . body]
```

Comment:
Ah, great change -- thanks!

I'm surprised it didn't through an error or warning, though ... pattern matching literal cons' has the form `` `(,head . ,tail)`` iirc. Ah, but that's for functions. Pattern matching in macros is a bit different ... I'd have to experiment more with that.